modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/dht/topology/PartitionEvictionOrderTest.java

```diff
@@ -50,6 +51,9 @@
  * Tests that {@link CacheRebalanceMode#SYNC} caches are evicted at first.
  */
 public class PartitionEvictionOrderTest extends GridCommonAbstractTest {
+    /** Condition check. */
+    boolean condCheck;
```

Comment:
Are you sure that there's happen-before between
condCheck = GridTestUtils.waitForCondition(() -> queue.size() == expEvictQueueSize, 100_000); **(system thread)**
and 
assertTrue(condCheck); **(test-runner thread)**
?